### PR TITLE
Support the GitHub ping event

### PR DESCRIPTION
**Pull request by hellcp-work**
This should make GitHub display a green status next to ping events when first creating webhooks in repos.

# To verify:
1. Set up testing according to [instructions on the wiki](https://github.com/openSUSE/open-build-service/wiki/Better-SCM-CI-Integration-Testing)
2. Send out a ping request
3. See a green response in OBS (and/or GitHub depending on the setup)

### CODE REVIEW

File: src/api/app/models/token/workflow.rb [hunk 1]
```diff
@@ -53,13 +56,17 @@ def validation_errors
...
-      @workflows.each do |workflow|
+      @workflows.try(:each) do |workflow|
         error_messages << workflow.errors.full_messages unless workflow.valid?
       end
 
       error_messages.flatten
     end
   end
+
+  def ping?
```
Comment (dmarcoux) [inline src/api/app/models/token/workflow.rb:67]: This should be a method in the [SCMWebhook](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/scm_webhook.rb) class. It doesn't belong here.

**Review by dmarcoux — DISMISSED** [review]
This should be covered with specs.

What is the payload for this webhook event like? Do you have an example?

File: src/api/app/models/token/workflow.rb [hunk 2]
```diff
@@ -25,14 +25,17 @@ def call(options)
...
     workflow_run.update(response_url: @scm_webhook.payload[:api_endpoint])
-    yaml_file = Workflows::YAMLDownloader.new(@scm_webhook.payload, token: self).call
-    @workflows = Workflows::YAMLToWorkflowsService.new(yaml_file: yaml_file, scm_webhook: @scm_webhook, token: self, workflow_run: workflow_run).call
+    # We can't get the yaml from a ping event, because the payload doesn't contain all the necessary information
+    unless ping?
+      yaml_file = Workflows::YAMLDownloader.new(@scm_webhook.payload, token: self).call
+      @workflows = Workflows::YAMLToWorkflowsService.new(yaml_file: yaml_file, scm_webhook: @scm_webhook, token: self, workflow_run: workflow_run).call
+    end
 
     return validation_errors unless validation_errors.none?
```
Comment (dmarcoux) [inline src/api/app/models/token/workflow.rb:34]: `validation_errors` is checking for validation errors in workflows, so this part of the validation should be skipped.

**Review by dmarcoux — APPROVED** [review]

File: src/api/spec/models/token/workflow_spec.rb [hunk 3]
```diff
@@ -159,5 +159,47 @@
...
+        allow(SCMStatusReporter).to receive(:new).and_return(proc { true })
+      end
+
+      subject { workflow_token.call(workflow_run: workflow_run, scm_webhook: scm_extractor.call) }
+
+      it 'returns before checking for validation errors' do
+        expect(subject).to be_empty
+      end
+
+      it { expect { subject }.to change(workflow_token, :triggered_at) & change(workflow_run, :response_url).to('https://api.github.com') }
```
Comment (eduardoj) [inline src/api/spec/models/token/workflow_spec.rb:197]:
```suggestion
      it { expect { subject }.to change(workflow_token, :triggered_at).and change(workflow_run, :response_url).to('https://api.github.com') }
```

There are some other parts of our code where we use the `&` operator. But we should make use of `and` to chain expectations. See https://relishapp.com/rspec/rspec-expectations/v/3-12/docs/compound-expectations#use-%60and%60-to-chain-expectations

This suggestion, still not tested, should also work.

**Review by eduardoj — APPROVED** [review]

**Pull request merged**